mycore-iview2/src/main/java/org/mycore/iview2/backend/MCRPDFThumbnailJobAction.java

```diff
@@ -0,0 +1,82 @@
+package org.mycore.iview2.backend;
+
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Locale;
+
+import javax.imageio.ImageIO;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.mycore.datamodel.metadata.MCRDerivate;
+import org.mycore.datamodel.metadata.MCRMetadataManager;
+import org.mycore.datamodel.metadata.MCRObjectID;
+import org.mycore.datamodel.niofs.MCRPath;
+import org.mycore.imagetiler.MCRImage;
+import org.mycore.iview2.frontend.MCRPDFTools;
+import org.mycore.iview2.services.MCRIView2Tools;
+import org.mycore.services.queuedjob.MCRJob;
+import org.mycore.services.queuedjob.MCRJobAction;
+
+public class MCRPDFThumbnailJobAction extends MCRJobAction {
+
+    public MCRPDFThumbnailJobAction() {
+    }
+
+    public MCRPDFThumbnailJobAction(MCRJob job) {
+        super(job);
+    }
+
+    public static final String DERIVATE_PARAMETER = "derivate";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5425159)